src/common/transformations/src/transformations/common_optimizations/loop_inputs.cpp

```diff
@@ -0,0 +1,117 @@
+
+        for (const auto& input_description : loop->get_input_descriptions()) {
+            if (const auto merged_input_desc =
+                    dynamic_pointer_cast<ov::op::util::MultiSubGraphOp::MergedInputDescription>(input_description)) {
+                if (get_parent_param(body_results[merged_input_desc->m_body_value_index])) {
+                    new_loop->set_invariant_input(body_params[merged_input_desc->m_body_parameter_index],
+                                                  loop_input_values[merged_input_desc->m_input_index]);
+                } else {
+                    new_loop->set_merged_input(body_params[merged_input_desc->m_body_parameter_index],
+                                               loop_input_values[merged_input_desc->m_input_index],
+                                               body_results[merged_input_desc->m_body_value_index]);
+                }
+            } else if (const auto invariant_input_desc =
+                           dynamic_pointer_cast<ov::op::util::MultiSubGraphOp::InvariantInputDescription>(
+                               input_description)) {
+                new_loop->set_invariant_input(body_params[invariant_input_desc->m_body_parameter_index],
+                                              loop_input_values[invariant_input_desc->m_input_index]);
+            } else if (const auto sliced_input_desc =
+                           dynamic_pointer_cast<ov::op::util::MultiSubGraphOp::SliceInputDescription>(
+                               input_description)) {
+                new_loop->set_sliced_input(body_params[sliced_input_desc->m_body_parameter_index],
+                                           loop_input_values[sliced_input_desc->m_input_index],
+                                           sliced_input_desc->m_start,
+                                           sliced_input_desc->m_stride,
+                                           sliced_input_desc->m_part_size,
+                                           sliced_input_desc->m_end,
+                                           sliced_input_desc->m_axis);
+            } else {
+                /* unknown input description type
+                 * this could only happen if new input description type was added after this transformation
+                 * written
+                 */
+                return false;
+            }
+        }
+
+        std::unordered_map<size_t, ov::Output<ov::Node>> loop_inputs;  // body_param->instance_id -> loop_input
+        for (const auto& input_description : loop->get_input_descriptions()) {
+            loop_inputs.emplace(body_params[input_description->m_body_parameter_index]->get_instance_id(),
+                                loop_input_values[input_description->m_input_index]);
+        }
+
+        ov::OutputVector new_loop_outputs;
+        for (const auto& output_description : loop->get_output_descriptions()) {
+            int64_t iteration = -1;
```

Comment:
we don't need to create it multiple times
Please move it above or use the value in the function call